hedera-node/src/main/java/com/hedera/services/ledger/AccountsCommitInterceptor.java

```diff
@@ -0,0 +1,67 @@
+package com.hedera.services.ledger;
+
+import com.hedera.services.context.SideEffectsTracker;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hederahashgraph.api.proto.java.AccountID;
+
+import java.util.List;
+import java.util.Map;
+
+public class AccountsCommitInterceptor implements
+		CommitInterceptor<AccountID, MerkleAccount, AccountProperty> {
+
+	// The tracker this interceptor should use for previewing changes. The interceptor is NOT
+	// responsible for calling reset() on the tracker, as that will be done by the client code.
+	private SideEffectsTracker sideEffectsTracker;
+	
+	public void setSideEffectsTracker(final SideEffectsTracker sideEffectsTracker) {
+		this.sideEffectsTracker = sideEffectsTracker;
+	}
+
+	/**
+	 * Accepts a pending change set, including creations and removals.
+	 *
+	 * @throws IllegalStateException if these changes are invalid
+	 */
+	@Override
+	public void preview(final List<MerkleLeafChanges<AccountID, MerkleAccount, AccountProperty>> changesToCommit) {
+		long sum = 0;
```

Comment:
It might be more readable if the `SideEffectsTracker` kept this `sum` internally (updating it with each call to `trackHbarChange()`, and setting `sum = 0` in `reset()`).

Then our consistency check would read like,
```
if (sideEffectsTracker.netHbarChange() != 0) {
    throw new IllegalStateException("...");
}
```